main.go

```diff
@@ -1684,8 +1684,15 @@ func main() {
 			usage(command)
 			os.Exit(1)
 		}
-		if options.Target == "" && filepath.Ext(outpath) == ".wasm" {
-			options.Target = "wasm"
+		if options.Target == "" && options.GOARCH == "wasm" {
+			switch options.GOOS {
+			case "js":
+				options.Target = "wasm"
+			case "wasip1":
+				options.Target = "wasip1"
+			case "wasip2":
+				options.Target = "wasip2"
+			}
```

Comment:
Perhaps we should add a default case here with an error message?
Something like (pseudocode-ish):

```go
case "":
    return errors.New("GOOS is unset but GOARCH is set to wasm. Please set GOOS to wasm, wasip1, or wasip2.")
default:
    return errors.New(fmt.Sprintf("Unknown GOOS=%s for GOARCH=wasm. Please set GOOS to wasm, wasip1, or wasip2.", options.GOOS)
```

Reply:
Thanks for the feedback. See my changes.